src/pages/AddItemForm.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { db } from '../lib/firebase';
+import { addDoc, collection } from 'firebase/firestore';
+// const formStyle = {
+//     display: 'block',
+//   };
+async function addToDb(item_name, purchase_interval, user_token) {
+  try {
+    const docRef = await addDoc(collection(db, 'groceries'), {
+      item_name: item_name,
+      purchase_interval: purchase_interval,
+      user_token: user_token,
+      last_purchased_date: null,
+    });
+
+    console.log('Document written with ID: ', docRef.id);
+  } catch (e) {
+    console.error('Error adding document: ', e);
+  }
+}
+
+class AddItemForm extends React.Component {
+  constructor(props) {
```

Comment:
Most people have moved away from using class based components in React and the need for a `constructor` needing to bind methods to this was removed quite some time ago. 

I'd recommend refactoring this to a functional component if you'd like to show it off to potential employers.

The refactor would be pretty straightforward and would look something like:

```javascript
const AddForm = (props) => {
   const [itemName, setItemName] = useState('');
   const [purchaseInterval, setPurchaseInterval]= useState('7');
 
   const handleItemNameChange = (e) => setItemName(e.target.value);
   const handleRadioChange (e) => setPurchaseInterval(e.target.value);

   const handleSubmit = (e) => {
      e.preventDefault();
      addToDb(
         itemName,
         purchaseInterval,
         props.token,
    );
   }
   return <form ... // remove this. from everything
}
```